internal/clients/grpc/connect.go

```diff
@@ -45,20 +50,42 @@ func NewClientHandler(c *Client) ClientHandler {
 
 // Connect creates new gRPC client and connect to gRPC server.
 func Connect(cfg Config) (*gogrpc.ClientConn, bool, error) {
-	var opts []gogrpc.DialOption
+	opts := []gogrpc.DialOption{
+		gogrpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+	}
 	secure := false
 	tc := insecure.NewCredentials()
 
-	if cfg.ClientTLS && cfg.CACerts != "" {
-		var err error
-		tc, err = credentials.NewClientTLSFromFile(cfg.CACerts, "")
+	if cfg.ServerCAFile != "" {
+		tlsConfig := &tls.Config{}
+
+		// Loading root ca certificates file
+		rootCA, err := ioutil.ReadFile(cfg.ServerCAFile)
```

Comment:
`ioutil.ReadFile` is deprecated as of 1.16

```suggestion
		rootCA, err := os.ReadFile(cfg.ServerCAFile)
```